### ISSUE TITLE
Vectorize _build_reindex_and_keep_tables in vocab embedding

### ISSUE THREAD

**Issue by prashantgupta24** [torch-spyre/spyre-inference#724]
## Summary

`_build_reindex_and_keep_tables` in `SpyreVocabParallelEmbedding` currently builds the reindex and keep lookup tables one vocabulary index at a time in a Python loop. For large vocabularies this can take seconds during model loading.

## Location

- File: `spyre_inference/custom_ops/vocab_parallel_embedding.py`
- Method: `SpyreVocabParallelEmbedding._build_reindex_and_keep_tables`
- Lines: 67-83

https://github.com/torch-spyre/spyre-inference/blob/main/spyre_inference/custom_ops/vocab_parallel_embedding.py#L67-L83

## Current code

```python
def _build_reindex_and_keep_tables(self) -> tuple[torch.Tensor, torch.Tensor]:
    """Build the reindex and keep lookup tables in one pass."""
    vocab_size = self.num_embeddings
    reindex_table = torch.zeros(vocab_size, 2, dtype=torch.int64)
    keep_table = torch.zeros(vocab_size, 2, dtype=torch.float16)
    for i in range(vocab_size):
        masked_input, input_mask = get_masked_input_and_mask(
            torch.tensor([i], dtype=torch.int64),
            self.shard_indices.org_vocab_start_index,
            self.shard_indices.org_vocab_end_index,
            self.shard_indices.num_org_vocab_padding,
            self.shard_indices.added_vocab_start_index,
            self.shard_indices.added_vocab_end_index,
        )
        reindex_table[i, 0] = masked_input.item()
        keep_table[i, 0] = 0.0 if input_mask.item() else 1.0
    return reindex_table, keep_table
```

## Suggested improvement

Vectorize the loop by calling `get_masked_input_and_mask` once on a `[vocab_size]` tensor if the upstream helper supports batched input, or replace the Python iteration with `torch.arange` + masking/indexing operations. This should noticeably reduce model-load time for large vocabularies.

## Tags

enhancement
